### ISSUE TITLE
Provide debugging context in sgrep error messages

### ISSUE THREAD

**Issue by nbrahms** [returntocorp/semgrep#383]
Currently, sgrep error messages can be difficult for a user to leverage.

Ideally, sgrep error messages would include the following context items:
- [ ] If the error is a parse error in the pattern, a parse error in the code, or an internal error within sgrep.
- [ ] If the error is a parse error, indicate which token(s) caused the parse issue, and which token(s) were expected.
- [ ] If the error is a parse error, provide surrounding AST context.
- [ ] If the error is an internal error, sgrep should emit sufficient context so that an sgrep developer can reproduce the issue (for example, which attempted match class fails, the input pattern, and the input program code or filename, in addition to sgrep version and code point that triggered the error).

**Comment by dlukeomalley** [comment]
Blocked on https://github.com/returntocorp/sgrep/issues/383

**Comment by nbrahms** [comment]
@dlukeomalley which issue is this blocked on?

**Comment by dlukeomalley** [comment]
Not blocked on any issues. About 19 hours ago I accidentally added and then removed the blocked Label.

**Comment by nbrahms** [comment]
cc. @mschwager

**Comment by ievans** [comment]
@nbrahms I think these are mostly done, and this issue is maybe a dupe of https://github.com/returntocorp/semgrep/issues/384?

**Comment by ievans** [comment]
#1-3 are done or tracked in https://github.com/returntocorp/semgrep/pull/1732
#4 is superseded by https://github.com/returntocorp/semgrep/issues/1652
